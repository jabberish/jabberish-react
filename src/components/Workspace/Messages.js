import React from 'react';
import PropTypes from 'prop-types';

import Message from './Message';

import { List, Paper } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';


const useStyles = makeStyles(theme => ({
  paper: {
    display: 'flex',
    flexDirection: 'column',
    height: '100%',
    width: '100%',
  },
  toolbar: theme.mixins.toolbar,
}));

const Messages = ({ messagesData }) => {
  const classes = useStyles();
  return (
    <Paper className={classes.paper}>
      <List>
        {messagesData.map(messageData => (
          <Message key={messageData._id} messageData={messageData} />
        ))}
      </List>
      <div className={classes.toolbar} />
    </Paper>
  );
};

Messages.propTypes = {
  messagesData: PropTypes.array.isRequired
};

export default Messages;
