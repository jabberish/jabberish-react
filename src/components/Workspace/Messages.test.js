/* eslint-disable no-console */
import React from 'react';
import { shallow } from 'enzyme';
import Messages from './Messages';

describe('Messages component', () => {
  it('renders Messages', () => {
    const wrapper = shallow(<Messages messagesData={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
