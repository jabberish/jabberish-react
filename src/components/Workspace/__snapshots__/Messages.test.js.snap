// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages component renders Messages 1`] = `
<WithStyles(ForwardRef(Container))
  className="makeStyles-container-1"
  maxWidth="false"
>
  <WithStyles(ForwardRef(List))
    className="makeStyles-list-2"
  >
    <div
      id="scroll-bottom"
      style={
        Object {
          "clear": "both",
          "float": "left",
        }
      }
    />
  </WithStyles(ForwardRef(List))>
</WithStyles(ForwardRef(Container))>
`;
